pkg/network/network_test.go

```diff
@@ -0,0 +1,66 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Network", func() {
+	Describe("Internal Functions", func() {
+		It("should NOT return an error because kube mac pool can be nil", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should NOT return an error because both or none of the KubeMacPool can be configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(0))
+		})
+
+		It("should return an error because only start range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "02:00:00:00:00:00", EndPoolRange: ""}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
+		})
+
+		It("should return an error because only end range is configured", func() {
+			clusterConfig := &opv1alpha1.NetworkAddonsConfigSpec{
+				KubeMacPool: &opv1alpha1.KubeMacPool{StartPoolRange: "", EndPoolRange: "02:00:00:FF:FF:FF"}}
+			errorList := validateKubeMacPool(clusterConfig)
+			Expect(len(errorList)).To(Equal(1))
+			Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"))
```

Comment:
To make test logs more readable, please add explanation **what** failed to each Expect. e.g.:

```
Expect(errorList[0].Error()).To(Equal("both or none of the KubeMacPool ranges needs to be configured"), "Validation failed with unexpected error message: %s", errorList[0].Error())
```

Similar in other tests. It really makes a big difference while working with failing tests.

Reply:
Hi @phoracek . Thank you for the comments. 
The error message in errorList[0] would be quite similar to the explanation that is in the "It" statement.
would you like me to add that anyway?
Thanks.

`	It("should return an error because only the end range is configured", func() {
`

Reply:
Good point. You are right, that if there is only a single Except within It block, it is not really needed. Please add this "explanation" message only in cases where it would be valuable to know, what the value is. e.g. if the value is unexpected, print it in Exception message.

Reply:
Thanks. Will do.